app/controllers/posts_controller.rb

```diff
@@ -8,10 +8,27 @@ def index
   def show
     id = params[:id].to_i
     author_id = params[:author_id].to_i
-    @post_author = User.find(author_id).name
+    @author = User.find(author_id)
+    @post_author = @author.name
     @post = Post.find(id)
     @likes = @post.likes_counter
     @comments_count = @post.comments_counter
     @comments = Comment.where(post_id: id)
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    title = params[:post][:title]
+    text = params[:post][:text]
+    post = Post.new(author_id: current_user.id, title:, text:, comments_counter: 0, likes_counter: 0)
+
+    if post.save
+      redirect_to "/users/#{current_user.id}/posts", notice: 'Post created successfully!!'
+    else
+      render :new, notice: 'An error occured while creating the post!'
+    end
+  end
```

Comment:
[OPTIONAL]
- I think it would be better to have the private params method here for your parameters. Here is an example.
```
    private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```
You can use it instead of individually passing each parameter.